packages/react/src/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,211 @@
+import React, {Children, useEffect, useRef, useState} from 'react'
+import clsx from 'clsx'
+import {useId} from '@reach/auto-id'
+import {getAnchoredPosition} from '@primer/behaviors'
+import type {AnchorSide, AnchorAlignment} from '@primer/behaviors'
+// @oddbird/popover-polyfill/fn import is mysteriously failing in eslint despite resolving correctly in the build
+// eslint-disable-next-line import/no-unresolved
+import {isSupported, apply} from '@oddbird/popover-polyfill/fn'
+import {useProvidedRefOrCreate} from '../hooks/useRef'
+
+/** * Main Stylesheet (as a CSS Module) */
+import styles from './Tooltip.module.css'
+
+type TooltipDirection = 'n' | 'e' | 's' | 'w'
+export type TooltipProps = React.PropsWithChildren<{
+  direction?: TooltipDirection
+  id?: string
+  text: string
+  type?: 'label' | 'description'
+  children?: React.ReactNode
```

Comment:
As you're already using `PropsWithChildren`, I don't think this manual declaration is necessary?

Reply:
Strangely, I'm getting TypeScript errors when I remove the `children` property from the type. I'm just going to remove `React.PropsWithChildren`.